### PR TITLE
fix naming issues in Connectables.map

### PR DESCRIPTION
**Pull request by pettermahlen**
- it's not really a 'map' but a 'contramap' - 'convert' is probably a better naming choice
- rename the input parameters to something more generic than M, E, V

### CODE REVIEW

**Comment by pettermahlen** [comment]
Humm.. I named them `F` to `T`, because that's what the user-supplied function does, but I guess that's the `contra` in `contramap`, so that's the wrong way to look at things. I have a problem with calling the method `contramap` because the results you get when googling for that are very hard to understand. That name will make the method hard to discover because you will only know what it means/that you might want to look for it if you're already pretty much hard-core FP.

Even if this is actually an example of the general concept of contravariant functors, I think we want this for a very specific use case here in Mobius, and I feel like we should possibly just mention the fact that it is a contravariant functor in the javadocs rather than call the method `contramap`. IIUC, an example of doing that is `Ordering.by` in the Scala standard library (mentioned here http://igstan.ro/posts/2013-10-31-contravariant-functors-an-intuition.html). The `by` method is `contramap`, but has a different name.

So, I agree that the `F` and `T` type names are bad, but not (yet) about calling the method `contramap`. :) I think it's more intuitive to think of it as converting a `Connectable<T, O>` to `Connectable<F, O>` with the current type names. And then `convert` is a better name, IMO. There may be even better ones.

**Comment by jeppes** [comment]
Perhaps the name `convert` would make more sense if F and T were switched, something like
`convert :: Connectable F O -> (T -> F) -> Connectable T O`
It is converting from a Connectable which takes F's to a Connectable that takes T's.

This ordering of the arguments also makes it clearer that it is the Connectable that is being converted.

**Comment by pettermahlen** [comment]
@anawara updated based on our discussion.

File: mobius-extras/src/main/java/com/spotify/mobius/extras/Connectables.java [hunk 1]
```diff
@@ -34,29 +34,72 @@ private Connectables() {
   }
 
   /**
-   * Apply a function to a {@link Connectable} in order to convert from one data type to another.
-   * This is useful for instance if you want your UI to use a subset or a transformed version of the
-   * full model used in the loop. The returned {@link Connectable} doesn't enforce a connection
+   * Convert a {@code Connectable<I, O>} to a {@code Connectable<J, O>} by applying a function from
+   * J to I for each J received, before invoking the {@code Connectable<I, O>}. This makes {@link
```
Comment (anawara) [inline mobius-extras/src/main/java/com/spotify/mobius/extras/Connectables.java]: I think you invoke the `Connection<I, O>` not the `Connectable`, right?

File: mobius-extras/src/main/java/com/spotify/mobius/extras/Connectables.java [hunk 2]
```diff
@@ -34,29 +34,72 @@ private Connectables() {
...
+   * Connectable} a <a
+   * href="https://hackage.haskell.org/package/contravariant-1.4.1/docs/Data-Functor-Contravariant.html">contravariant
+   * functor</a> in functional programming terms.
+   *
+   * <p>This is useful for instance if you want your UI to use a subset or a transformed version of
+   * the full model used in the loop. The returned {@link Connectable} doesn't enforce a connection
```
Comment (anawara) [inline mobius-extras/src/main/java/com/spotify/mobius/extras/Connectables.java]: nit: "used in the MobiusLoop"

File: mobius-extras/src/main/java/com/spotify/mobius/extras/Connectables.java (hunk 1 above)
Comment (anawara) [reply mobius-extras/src/main/java/com/spotify/mobius/extras/Connectables.java]: Creates a `Connectable<J, O>` that transforms all `J` into `I` before forwarding it to the actual `Connection<I>`

**Pull request merged**